### DISCUSSION TITLE
Sending data between webparts

### DISCUSSION THREAD

**Discussion by Nevinia210** [microsoft-search/pnp-modern-search#2262] [Q&A] [answered]
Hello Everyone,

Currently I'm working on a webpart that will show taxonomy tree to the user and after user click on some taxonomy, it will show results in  "PnP - Search Results" webpart
this is my code for "registering" webpart dynamic data provider, implementation of inherited IDynamicDataCallables methods and actual onchange event
```TypeScript
 protected onInit(): Promise<void> {
    this.initializeWebPartServices();
    this._environmentMessage = this._getEnvironmentMessage();
    this.loadTermStore();
    this.context.dynamicDataSourceManager.initializeSource(this);
    return super.onInit();
  }

  private initializeWebPartServices(): void {
    this.tokenService = this.context.serviceScope.consume<ITokenService>(TokenService.ServiceKey);
    this.webPartInstanceServiceScope = this.context.serviceScope.startNewChild();
    this.dynamicDataService = this.webPartInstanceServiceScope.createDefaultAndProvide(DynamicDataService.ServiceKey);
    this.dynamicDataService.dynamicDataProvider = this.context.dynamicDataProvider;
    this.webPartInstanceServiceScope.finish();
  }

  public getPropertyDefinitions(): ReadonlyArray<IDynamicDataPropertyDefinition> {
    return [
      {
        id: "SiteMapSearchWebPart",
        title: "Site Map Search WebPart"
      }
    ];
  }

  public getPropertyValue(propertyId: string) {

    return this.termQuery;
  }

 protected changeTermValue(e) {
    if (e.length == 0) {
      this.termQuery = "";

      GlobalSettings.setValue("inputQueryText", this.termQuery);
      this.context.dynamicDataSourceManager.notifyPropertyChanged("SiteMapSearchWebPart");
      this.context.dynamicDataSourceManager.notifySourceChanged();

      console.log("value unclicked");
    }
    else {
      this.termQuery = "sitemapproperty:\"" + e[0].key + "\"";

      GlobalSettings.setValue("inputQueryText", this.termQuery);
      this.context.dynamicDataSourceManager.notifyPropertyChanged("SiteMapSearchWebPart");
      this.context.dynamicDataSourceManager.notifySourceChanged();

      console.log("new value clicked: " + this.termQuery);
    }
  }
```

in following picture is configuration for webpart query template (from first page of configuration) and configuration of dynamic search source
![image](https://user-images.githubusercontent.com/42325625/182633492-97af876e-7b94-48be-91e5-095d8f94850f.png)

you can see that {inputQueryText} is blank text even when in that time i have selected item
Could you please help me to find where the issue is?

Thank you in advance for your help :)

**Comment by FranckyC** [comment, marked as answer]
Hi @Nevinia210, not sure what you are trying to do reusing `TokenService`, `DynamicDataService` and `GlobalSettings` in your WebPart, you don't really need any of that. If you want the output of your Web Part to be injected as a search term in the search results Web Part, when an user clicks on a term, simply following this example here https://docs.microsoft.com/en-us/sharepoint/dev/spfx/dynamic-data should do the trick.

**Reply by Nevinia210** [reply]
Hi @FranckyC ,
thanks for your answer, in my first version of code i did not have `TokenService`, `DynamicDataService` and `GlobalSettings` at all, this is just my code expanding to more complex stuff after fail result
as i followed all these i still did not get result that my queryModification property contains query that im sending to it, maybe i have some bug in a code?

**Reply by FranckyC** [reply]
In the query template field of the search results WP, try to use `{inputQueryText}` instead of `{searchTerms}` see if there is a difference. Otherwise, pretty hard to debug from here. To make sure, is `this.termQuery` a string?

**Reply by Nevinia210** [reply]
@FranckyC 
in query template field of the search result WP there is allready `{inputQueryText}`, its in the picture above, but i also tried that searchTerms and its no difference,
yes this.termQuery is a string

**Reply by FranckyC** [reply]
Do you have a repo with your code so I can test it and see what would be the issue?

**Reply by Nevinia210** [reply]
Hi @FranckyC ,
sorry for such a long reply, i was on vacation,
so after the vacaion i checked code again and it really match with the example that you sended to me, so i tried to debug it whole and find out that getPropertyValue is really called after change, but the value is not correct because i send to child component in react just a method and not method.bind(this)
this causes that method vas invoked in childs component and changed childs property and then in a parent component when value vas sended, it sent empty value,
so after all problem was somewhere else than in the connection of 2 webparts,
issue can be closed,
thanks for your replies
